docs/DeveloperGuide.md

```diff
@@ -6,33 +6,230 @@
 
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### The Overall UML Diagram:
 
+![UML Diagram](assets/images/mathGenius.png)    
```

Comment:
Is the (c) icon in UML diagram meant to be there? Please consider removing it. 
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/111263799/ed5f8488-fae4-4403-86b7-25f499f0ddbf)